### PR TITLE
feat(data): Implement Core Data DELETE /event/id/{id} V2 API

### PR DESCRIPTION
**Pull request by judehung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
#2834 

## What is the new behavior?
Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/default/delete_event_id__id_,
implement the V2 API for DELETE /event/id/{id}

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by judehung** [comment]
rebased

**Comment by judehung** [comment]
rebased

**Comment by judehung** [comment]
recheck

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by judehung** [comment]
rebased

File: internal/core/data/v2/controller/http/event.go [hunk 1]
```diff
@@ -126,6 +126,40 @@ func (ec *EventController) EventById(w http.ResponseWriter, r *http.Request) {
...
+	id := vars[contractsV2.Id]
+
+	var response interface{}
+	var statusCode int
+
+	// Delete the event
+	err := application.DeleteEventById(id, ec.dic)
+	if err != nil {
+		// Event not found is not a real error, so the error message should not be printed out
+		if errors.Kind(err) != errors.KindEntityDoesNotExist {
```
Comment (cloudxxx8) [inline internal/core/data/v2/controller/http/event.go:147]: delete action should log error anyway

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: internal/core/data/v2/controller/http/event.go (hunk 1 above)
Comment (judehung) [reply internal/core/data/v2/controller/http/event.go]: updated.

**Comment by cloudxxx8** [comment]
recheck

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**